uint8 status
float64 curvature
bool side